logger/logger.go

```diff
@@ -0,0 +1,118 @@
+package logger
+
+import (
+	"errors"
+	"math/rand"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var (
+	ErrEmptyService = errors.New("empty service name")
+
+	mu sync.Mutex
```

Comment:
Not sure if we should be using a global mutex here. This logger is intended to be used in very high volume settings and not sure if we would want to block the entire thread every time we log. 

Possible other solutions could be a more granular locking mechanism or buffering the logs and writing in chunks?

Curious to hear others' thoughts as to whether this mutex is good practice or not?

Reply:
It won't block the thread since the `lock` it's only called on the console writing, and since `goroutines` locks are design to block a portion of the code, this lock will not block the thread, it would only block the writings in the console (at least that's how it supposed to work). However, I'll add a note to make sure that everyone keeps an eye on it while we test this